examples/rsa-example.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <iostream>
+#include <mococrw/asymmetric_crypto_ctx.h>
+#include <mococrw/padding_mode.h>
+
+using namespace mococrw;
+
+std::vector<uint8_t> encrypt_rsa(const std::shared_ptr<AsymmetricPublicKey> &pubKey,
+                                 const std::shared_ptr<RSAEncryptionPadding> &padding,
+                                 const std::vector<uint8_t> &data)
+{
+    RSAEncryptionPublicKeyCtx rsaPubCtx(*pubKey, padding);
+
+    std::vector<uint8_t> ciphertext;
+    try {
+        ciphertext = rsaPubCtx.encrypt(data);
+    } catch (openssl::OpenSSLException &e) {
+        std::cerr << "Failure encrypting data." << std::endl;
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    } catch (mococrw::MoCOCrWException &e) {
+        std::cerr << "Failure in crypto engine: ";
+        std::cerr << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    return ciphertext;
+}
+
+std::vector<uint8_t> decrypt_rsa(const std::shared_ptr<AsymmetricPrivateKey> &privKey,
+                                 const std::shared_ptr<RSAEncryptionPadding> &padding,
+                                 const std::vector<uint8_t> &ciphertext) {
+    std::vector<uint8_t> plaintext;
+    /* Padding is optional. Default is OAEP with SHA256 and MGF1(SHA256) as mask generation function */
+    RSAEncryptionPrivateKeyCtx rsaPrivCtx(*privKey, padding);
+    try {
+        plaintext = rsaPrivCtx.decrypt(ciphertext);
+    }  catch (MoCOCrWException &e) {
+        std::cerr << "Error decrypting data." << std::endl;
+        std::cerr << e.what();
+        exit(EXIT_FAILURE);
+    }
+
+   return plaintext;
+}
+
+int main(void)
+{
+    auto keyPair = std::make_shared<AsymmetricKeypair>(AsymmetricKeypair::generateRSA());
+    std::vector<uint8_t> data = utility::fromHex("deadbeef");
+    DigestTypes oaepDigestType = DigestTypes::SHA512;
+    DigestTypes mgfDigestType = DigestTypes::SHA3_256;
+    std::string oaepLabel = "this is my awesome label";
```

Comment:
I like that you're showing that we can adjust all these things, but since your example is a bit uncommon (usually the same hash is used for OAEP and the MGF, the label is virtually never modified and always the default), I'd appreciate if you could add a short comment that this a very constructed example and you usually don't have to set these.